content/en/docs/concepts/scheduling-eviction/api-eviction.md

```diff
@@ -6,14 +6,61 @@ weight: 70
 
 {{< glossary_definition term_id="api-eviction" length="short" >}} </br>
 
-You can request eviction by directly calling the Eviction API 
-using a client of the kube-apiserver, like the `kubectl drain` command. 
-This creates an `Eviction` object, which causes the API server to terminate the Pod. 
+You can request eviction by calling the Eviction API directly, or programmatically
+using a client of the kube-apiserver, like the `kubectl drain` command. This
+creates an `Eviction` object, which causes the API server to terminate the Pod.
 
 API-initiated evictions respect your configured [`PodDisruptionBudgets`](/docs/tasks/run-application/configure-pdb/)
 and [`terminationGracePeriodSeconds`](/docs/concepts/workloads/pods/pod-lifecycle#pod-termination). 
 
+Using the API to create an Eviction object for a Pod is like performing a
+policy-controlled DELETE operation on the Pod. 
+
+## Calling the Eviction API
+
+You can use a [Kubernetes language client](/docs/tasks/administer-cluster/access-cluster-api/#programmatic-access-to-the-api)
+to access the Kubernetes API and create an `Eviction` object. To do this, you
+POST the attempted operation.
+
+Alternatively, you can attempt an eviction operation by accessing the API using
+`curl` or `wget`. 
+
+## How API-initiated eviction works
+
+When you attempt to create an `Eviction` object, the API responds in one of the
+following ways:
+
+* `200 OK`: the eviction is allowed and the Pod is deleted, similar to sending a
+  `DELETE` request to the Pod URL.
+* `429 Too Many Requests`: the eviction is not currently allowed because of the
+  configured PodDisruptionBudget. You may be able to attempt the eviction again
+  later.
+* `500 Internal Server Error`: the eviction is not allowed because there is a
+  misconfiguration, like if multiple PodDisruptionBudgets reference the same Pod.
+
+If the Pod you want to evict doesn't have a PodDisruptionBudget, the server always
+returns `200 OK` and allows the eviction.
+
+[[Need more information about the eviction object. Once it's created, what happens
+to cause the Pod to shut down? What control plane components work to get the job done?]]
```

Comment:
```suggestion
The control plane signals the eviction to the kubelet running that Pod; that
node's kubelet notices the signal and triggers pod shutdown.
```

We don't have to specify the control plane component involved. I think it's the API server but in the distant future it might not be.

Reply:
> What control plane component sends the eviction signal to the kubelet?

Maybe it's never built in to the control plane? Might be something like the [descheduler](https://github.com/kubernetes-sigs/descheduler), or `kubectl drain`.

Reply:
Aren't we describing here what happens **after** the eviction object has been created? How does that finally cause the kubelet to terminate the pod.

Reply:
Ah, right, yep. I think it's this:
the kubelet uses a [watch](https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes) for each Pod it's running, and sees that the Pod has been deleted (equivalent to an HTTP `DELETE`). When the kubelet finds it is running a Pod with no local static Pod and no remote Pod in the API either, the kubelet shuts down that Pod's containers and then the Pod itself.

We might want to migrate some of the details that currently live within https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/ into the Concepts section, and change that task page to link to the moved text.

Reply:
The “multiple” watches might be a single `GET` request using a field selector; I haven't looked at the code at all.

Reply:
So the Eviction object creation directly causes the Pod to get a deletion timestamp?

Reply:
> So the Eviction object creation directly causes the Pod to get a deletion timestamp?

I think: Yes, because the checks that apply (eg PodDisruptionBudget) happen at a phase that is logically earlier than the creation of the Eviction.

You don't actually find an Eviction object in the API as a result of a successful creation, but you could model it a bit like this:
- request eviction
- admission checks (include PodDisruptionBudget) succeed
- eviction at API level happens atomically
   - Eviction subresource created
   - Pod resource is marked for deletion
   - Eviction subresource removed as Pod is being deleted

The Pod API is complicated and several SIGs are involved in just that bit I've explained, so it might be wrong. How about we write that up and then see about getting a tech review?

Reply:
> We might want to migrate some of the details that currently live within 
> https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/ into the Concepts section, and change 
> that task page to link to the moved text.

I like that idea. I think it's out of scope for this PR but will be a good follow-up issue. 

> How about we write that up and then see about getting a tech review?

Will do.

Reply:
@sftim see if this looks any better?